src/services/review/ReviewRequestService.ts

```diff
@@ -481,47 +487,57 @@ export default class ReviewRequestService {
     sessionData: UserWithSiteSessionData,
     site: Site
   ): Promise<void> => {
-    const { isomerUserId: userId } = sessionData
-
-    const requestsViewed = await this.reviewRequestView.findAll({
-      where: {
-        siteId: site.id,
-        userId,
-      },
-    })
+    try {
+      const { isomerUserId: userId } = sessionData
+      await this.sequelize.transaction(async (transaction) => {
+        const requestsViewed = await this.reviewRequestView.findAll({
+          where: {
+            siteId: site.id,
+            userId,
+          },
+          transaction,
+        })
 
-    const allActiveRequests = await this.repository.findAll({
-      where: {
-        siteId: site.id,
-        // NOTE: Closed and merged review requests would not have an
-        // entry in the review request views table
-        reviewStatus: ["OPEN", "APPROVED"],
-      },
-    })
+        const allActiveRequests = await this.repository.findAll({
+          where: {
+            siteId: site.id,
+            reviewStatus: ["OPEN", "APPROVED"],
+          },
+          transaction,
+        })
 
-    const requestIdsViewed = requestsViewed.map(
-      (request) => request.reviewRequestId
-    )
-    const allActiveRequestIds = allActiveRequests.map((request) => request.id)
-    const requestIdsToMarkAsViewed = _.difference(
-      allActiveRequestIds,
-      requestIdsViewed
-    )
+        const requestIdsViewed = requestsViewed.map(
+          (request) => request.reviewRequestId
+        )
+        const allActiveRequestIds = allActiveRequests.map(
+          (request) => request.id
+        )
+        const requestIdsToMarkAsViewed = _.difference(
+          allActiveRequestIds,
+          requestIdsViewed
+        )
 
-    await Promise.all(
-      // Using map here to allow creations to be done concurrently
-      // But we do not actually need the result of the view creation
-      requestIdsToMarkAsViewed.map(async (requestId) =>
-        this.reviewRequestView.create({
-          reviewRequestId: requestId,
-          siteId: site.id,
-          userId,
-          // This field represents the user opening the review request
-          // itself, which the user has not done so yet at this stage.
-          lastViewedAt: null,
-        })
-      )
-    )
+        await Promise.all(
```

Comment:
actually i just realised this is because of duplicates right? we can just call `_.uniq(requestIdsToMarkAsViewed`. transaction seems abit heavy here tbh because the view status isn't super important (we don't really care if the notification is mistakenly shown as not viewed). also, i think it'll still run into the issue of duplicates, no?